core/src/test/java/org/mule/util/xmlsecurity/XMLSecureFactoriesPropertiesTestCase.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.util.xmlsecurity;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+import javax.xml.transform.TransformerFactory;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+public class XMLSecureFactoriesPropertiesTestCase extends AbstractMuleTestCase
+{
+
+    private final DefaultXMLSecureFactories defaultXMLSecureFactories = new DefaultXMLSecureFactories(false, false);
+    private final SchemaFactory schemaFactory = XMLSecureFactories.createDefault().getSchemaFactory("http://www.w3.org/2001/XMLSchema");
+    private final TransformerFactory transformerFactory = XMLSecureFactories.createDefault().getTransformerFactory();
+    private final Validator validatorWrapper = mock(Validator.class);
+    private final SchemaFactory schemaFactoryWrapper = mock(SchemaFactory.class);
+    private final TransformerFactory transformerFactoryWrapper = mock(TransformerFactory.class);
+
+    @Test
+    public void validatorProperties() throws Exception
+    {
+        SetPropertyAnswer setPropertyAnswer = new SetPropertyAnswer(schemaFactory.newSchema().newValidator());
+        doAnswer(setPropertyAnswer).when(validatorWrapper).setProperty(anyString(), anyObject());
+        defaultXMLSecureFactories.configureValidator(validatorWrapper);
+        assertThat(setPropertyAnswer.exception, is(nullValue()));
+        verify(validatorWrapper, times(2)).setProperty(anyString(), anyObject());
```

Comment:
How does this test validate that the correct properties are set?

Reply:
This mock object has an answer for the setProperty() method that invokes the real method and verifies any exception was thrown.